### PR TITLE
Added address to Sole trader no-nino audit event

### PR DESCRIPTION
**Pull request by Daveo-G**

### CODE REVIEW

File: it/uk/gov/hmrc/soletraderidentificationfrontend/controllers/BusinessVerificationControllerISpec.scala [hunk 1]
```diff
@@ -71,14 +71,17 @@ class BusinessVerificationControllerISpec extends ComponentSpecHelper
         stubRetrieveBusinessVerificationResultFromStub(testBusinessVerificationJourneyId)(OK, Json.obj("verificationStatus" -> "PASS"))
         stubStoreBusinessVerificationStatus(journeyId = testJourneyId, businessVerificationStatus = BusinessVerificationPass)(status = OK)
         stubRetrieveBusinessVerificationStatus(testJourneyId)(OK, Json.toJson(BusinessVerificationPass))
+        stubRetrieveSoleTraderDetails(testJourneyId)(OK, Json.toJson(testSoleTraderDetails(true)))
         stubRetrieveNino(testJourneyId)(OK, testNino)
+        stubRetrieveSAPostcode(testJourneyId)(NOT_FOUND)
         stubRetrieveSautr(testJourneyId)(OK, testSautr)
         stubRegister(testNino, testSautr)(OK, Registered(testSafeId))
         stubStoreRegistrationStatus(testJourneyId, Registered(testSafeId))(OK)
         stubAudit()
```
Comment (HarrNM) [inline it/uk/gov/hmrc/soletraderidentificationfrontend/controllers/BusinessVerificationControllerISpec.scala:80]: All these calls after the stubAudit() are for the audit service. With the new refactor we could remove a lot of these to make it a bit smaller. Could also be removed from the other IT specs that stub the audit as well

File: app/uk/gov/hmrc/soletraderidentificationfrontend/services/AuditService.scala [hunk 2]
```diff
@@ -74,26 +74,23 @@ class AuditService @Inject()(auditConnector: AuditConnector, soleTraderIdentific
 
   def auditSoleTraderJourney(journeyId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] = {
     for {
-      optFullName <- soleTraderIdentificationService.retrieveFullName(journeyId)
-      optDateOfBirth <- soleTraderIdentificationService.retrieveDateOfBirth(journeyId)
+      optSoleTraderRecord <- soleTraderIdentificationService.retrieveSoleTraderDetails(journeyId)
       optNino <- soleTraderIdentificationService.retrieveNino(journeyId)
```
Comment (HarrNM) [inline app/uk/gov/hmrc/soletraderidentificationfrontend/services/AuditService.scala:78]: The nino also gets pulled back in the optSoleTraderRecord so could we remove this extra nino call as well maybe?

File: it/uk/gov/hmrc/soletraderidentificationfrontend/controllers/BusinessVerificationControllerISpec.scala (hunk 1 above)
Comment (Daveo-G) [reply it/uk/gov/hmrc/soletraderidentificationfrontend/controllers/BusinessVerificationControllerISpec.scala:80]: Fixed

File: app/uk/gov/hmrc/soletraderidentificationfrontend/services/AuditService.scala (hunk 2 above)
Comment (Daveo-G) [reply app/uk/gov/hmrc/soletraderidentificationfrontend/services/AuditService.scala]: fixed

**Pull request merged**